### ISSUE TITLE
Possible bug in sampling code

### ISSUE THREAD

**Issue by nreHieW** [FoundationVision/VAR#119]
Hi in the following generation function:
```python
@torch.no_grad()
    def autoregressive_infer_cfg(
        self, B: int, label_B: Optional[Union[int, torch.LongTensor]],
        g_seed: Optional[int] = None, cfg=1.5, top_k=0, top_p=0.0,
        more_smooth=False,
    ) -> torch.Tensor:   # returns reconstructed image (B, 3, H, W) in [0, 1]
```

In this function, the model only receives `next_token_map` as input each forward pass instead of the entire sequence of all the other token maps before it. For example for patch size (1, 2, 3), the model takes in sequences of length 1 4 and 9 each time instead of the accumulation (1, 5, 14) 

In contrast, training is done with all the token maps concatenated into a single sequence and trained with the block causal mask.

is this intended?

**Comment by Kumbong** [comment]
KV caching is used to store the previous ones
